layer/layer.go

```diff
@@ -188,6 +188,8 @@ type Store interface {
 	GetRWLayer(id string) (RWLayer, error)
 	GetMountID(id string) (string, error)
 	ReleaseRWLayer(RWLayer) ([]Metadata, error)
+	LockRWLayer()
+	UnlockRWLayer()
```

Comment:
ping @dmcgowan 

I'm not sure this interface change is a good idea. It feels like it breaks the abstraction.

It's been awhile since I looked at this code, but would it make sense for `RWLayer`'s `TarStream` method to lock the layer while the stream is in progress?